cluster-autoscaler/core/filter_out_schedulable.go

```diff
@@ -0,0 +1,68 @@
+package core
+
+import (
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/metrics"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/pods"
+	"k8s.io/klog"
+)
+
+type filterOutSchedulablePodListProcessor struct{}
+
+func NewFilterOutSchedulablePodListProcessor() pods.PodListProcessor {
+	return &filterOutSchedulablePodListProcessor{}
+}
+
+// Process filters out pods which are schedulable from list of unschedulable pods.
+func (filterOutSchedulablePodListProcessor) Process(
+	context *context.AutoscalingContext,
+	unschedulablePods []*apiv1.Pod, allScheduledPods []*apiv1.Pod,
+	allNodes []*apiv1.Node, readyNodes []*apiv1.Node) ([]*apiv1.Pod, []*apiv1.Pod, error) {
+	// We need to check whether pods marked as unschedulable are actually unschedulable.
+	// It's likely we added a new node and the scheduler just haven't managed to put the
+	// pod on in yet. In this situation we don't want to trigger another scale-up.
+	//
+	// It's also important to prevent uncontrollable cluster growth if CA's simulated
+	// scheduler differs in opinion with real scheduler. Example of such situation:
+	// - CA and Scheduler has slightly different configuration
+	// - Scheduler can't schedule a pod and marks it as unschedulable
+	// - CA added a node which should help the pod
+	// - Scheduler doesn't schedule the pod on the new node
+	//   because according to it logic it doesn't fit there
+	// - CA see the pod is still unschedulable, so it adds another node to help it
+	//
+	// With the check enabled the last point won't happen because CA will ignore a pod
+	// which is supposed to schedule on an existing node.
+
+	// Some unschedulable pods can be waiting for lower priority pods preemption so they have nominated node to run.
+	// Such pods don't require scale up but should be considered during scale down.
+	unschedulablePods, unschedulableWaitingForLowerPriorityPreemption := filterOutExpendableAndSplit(unschedulablePods, context.ExpendablePodsPriorityCutoff)
```

Comment:
This was not created in this PR, but reading it now I think it's actually wrong. Why do we need to keep track of `unschedulableWaitingForLowerPriorityPreemption`? My intuition is we want to just treat them as if they were scheduled on nominated node. This is reflected in code - literally all we ever do with this list is append those pods to allScheduled (both in scale down and filterOutSchedulableBy*). 
Except we only don't append them to allScheduled when calling TryToScaleDown. This make some sense - if they're not really scheduled we shouldn't drain them. But the same could be said about any other pods artificially scheduled by PLPs. So it feels like a more correct thing to do would be:
- save original list of scheduled pods (and pass that to TryToScaleDown). 
- make filterOutExpendableAndSplit a separate PLP (just add unschedulableWaitingForLowerPriorityPreemption to scheduledPods).
WDYT?